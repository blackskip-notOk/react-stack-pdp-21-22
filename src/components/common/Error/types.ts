import { AxiosError } from 'axios';

export interface ErrorMessageProps {
	error: AxiosError | Error;
}
