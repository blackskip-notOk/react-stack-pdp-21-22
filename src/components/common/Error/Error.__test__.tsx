import { it } from 'vitest';

it('Error message should renders correctly', () => {
	const sum = (a: number, b: number) => a + b;

	expect(sum(2, 2)).toBe(4);
});
