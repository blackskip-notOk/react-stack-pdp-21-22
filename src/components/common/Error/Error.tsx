import { FC } from 'react';
import { ErrorMessageProps } from './types';

export const ErrorMessage: FC<ErrorMessageProps> = ({ error }) => {
	return (
		<div role='alert'>
			{error.name && <div>{error.name}</div>}
			<pre>{error.message}</pre>
		</div>
	);
};
