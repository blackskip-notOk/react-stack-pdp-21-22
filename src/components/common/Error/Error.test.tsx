import { ErrorMessage } from './Error';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('ErrorMessage.tsx', () => {
	const error = new Error('test error');

	test('Error message should show correct message', () => {
		render(<ErrorMessage error={error} />);

		expect(screen.getAllByText(/test error/i)).toBeDefined();
	});

	test('Error message should renders correctly', () => {
		const wrapper = renderer.create(<ErrorMessage error={error} />).toJSON();

		expect(wrapper).toMatchSnapshot();
	});
});
