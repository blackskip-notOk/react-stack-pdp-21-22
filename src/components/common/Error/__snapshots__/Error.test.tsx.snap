// Vitest Snapshot v1

exports[`ErrorMessage.tsx > Error message should renders correctly 1`] = `
<div
  role="alert"
>
  <div>
    Error
  </div>
  <pre>
    test error
  </pre>
</div>
`;
